### ISSUE TITLE
API function for new QCEvaluation lambda function

### ISSUE THREAD

**Issue by dbirman** [AllenNeuralDynamics/aind-data-access-api#93]
# User story
As a user I want to be able to upload QCEvaluations through a function in the data-access-api

# Acceptance criteria
- [ ] Able to upload a QCEvaluation object

# Sprint Ready Checklist 
- [ ] 1. Acceptance criteria defined 
- [ ] 2. Team understands acceptance criteria 
- [ ] 3. Team has defined solution / steps to satisfy acceptance criteria 
- [ ] 4. Acceptance criteria is verifiable / testable 
- [ ] 5. External / 3rd Party dependencies identified
- [ ] 6. Ticket is prioritized and sized

# Notes
Example of the full code needed:

```
import boto3
import requests
from aws_requests_auth.aws_auth import AWSRequestsAuth
 
session = boto3.Session()
credentials = session.get_credentials()
host = "api.allenneuraldynamics-test.org"
 
auth = AWSRequestsAuth(
  aws_access_key=credentials.access_key,
  aws_secret_access_key=credentials.secret_key,
  aws_token=credentials.token,
  aws_host="api.allenneuraldynamics-test.org",
  aws_region='us-west-2',
  aws_service='execute-api'
)
url = f"https://{host}/v1/add_qc_evaluation"
qc_eval = {"key": "value"}
post_request_content = {"data_asset_id": "record uuid", "qc_evaluation": qc_eval}
response = requests.post(url=url, auth=auth, json=post_request_content)
```

**Comment by dbirman** [comment]
> [@dbirman](https://github.com/dbirman) the qc lambda function accepts both single and list of evals. We can either have 1 method, e.g. `MetadataDbClient.add_qc_evaluation` that accepts `qc_eval` as list of single dictionary, or we can have a separate `MetadataDbClient.add_qc_evaluations` method. I have a slight preference for the latter. What do you think?

I'm partial to functions that accept a type or a list[type], sorry that we disagree!

Also flagging that the V2 implementation of this will be fairly different since it will need a new lambda function (to accept metrics, instead of evaluations) and then the V2 version will just be related to metrics. I think we want this V1 function though either way.

**Comment by dbirman** [comment]
> [@dbirman](https://github.com/dbirman) It might be more future-proof to accept `qc_contents: Dict[Any]` then. The user must then provide `{"qc_evaluation": qc_eval}`, or `{"qc_evaluation": list_of_qc_eval}`, or in the future some `{"qc_metrics": qc_metrics}`.

Yeah that would future proo,  but I think because this is something the scientists interact with we should keep it as simple as possible. `QCEvaluation | list[QCEvaluation` and then for v2 `QCMetric | list[QCMetric]` is going to be a lot more understandable for them then needing to go into the documentation to learn that you have to pass a keyed dictionary and that the key choice depends on the schema version you're uploading from.

**Comment by helen-m-lin** [comment]
Let's leave serialization of QC objects to a later issue if needed- any aind-data-schema dependent methods should be added in [helpers]. The method added to the MetadataDbClient should only handle signing and sending the request to the API gateway.
